tests/test_env.py

```diff
@@ -100,7 +101,9 @@ def install(self, requirements):
 
 
 @pytest.mark.isolated
-@pytest.mark.skipif(sys.version_info > (3, 6, 5), reason='inapplicable')
-def test_default_pip_is_upgraded_on_python_3_6_5_and_below():
+def test_default_pip_is_never_too_old():
     with build.env.IsolatedEnvBuilder() as env:
-        assert not subprocess.check_output([env.executable, '-m', 'pip', '-V']).startswith(b'pip 9.0')
+        version = subprocess.check_output(
+            [env.executable, '-c', 'import pip; print(pip.__version__)'], universal_newlines=True
```

Comment:
I don't think pip supports any format other than CLI. Instead you should do ``python -m pip --version`` and then do a split(' ')[1] on the output.

Reply:
It might be worthwhile to ask ono pip’s issue tracker about this being an exception. It’s not like there are many obvious ways to read that value…

Reply:
They removed all public interface with the release of 10.0, including the interface the `pip` script used, yet this stayed.

Reply:
I've asked here: https://discuss.python.org/t/is-is-safe-to-use-pip-version/7155

Reply:
It's allowed. :)